benches/benchmark.rs

```diff
@@ -0,0 +1,207 @@
+use std::collections::{BTreeMap, HashMap};
```

Comment:
Could you post the results from these once you feel like they're in a decent place?

Reply:
Will do. Done for today though. In their current state compression with unlimited range against no compression is about 20% faster for btreemap and about 35% faster for hashmap, with limited compression falling somewhere in between. Considering that due to the higher lookup-cost btreemap should be the one benefitting more there definitely is something fishy going on.